### PR TITLE
added azure blob bucket and fix bug

### PR DESCRIPTION
**Pull request by Lcch**
- added azure blob bucket
- fix mapper bucket bug
- master rand.Seed

### CODE REVIEW

File: bucket/bucket.go [hunk 1]
```diff
@@ -52,6 +52,37 @@ func NewBucket(bucketType string, config map[string]interface{}) (Bucket, error)
...
+		}
+		blobServiceBaseUrl, ok := config["blobServiceBaseUrl"]
+		if !ok {
+			return nil, fmt.Errorf("blobServiceBaseUrl is not provided")
+		}
+		apiVersion, ok := config["apiVersion"]
+		if !ok {
+			return nil, fmt.Errorf("apiVersion is not provided")
+		}
+		useHttps, ok := config["useHttps"]
```
Comment (comzyh) [inline bucket/bucket.go]: we can have default value on  `apiVersion `,  `useHttps` and `directory`

Comment (Lcch) [inline bucket/bucket.go]: done

File: bucket/azureblob.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	"github.com/Azure/azure-sdk-for-go/storage"
+)
+
+// max block size for azure blob
+const maxBlockSize = 50 * 1024 * 1024
+
+// AzureBlobBucket use azure blob
+type AzureBlobBucket struct {
+	container *storage.Container
+	directory string
```
Comment (Lcch) [inline bucket/azureblob.go]: done

File: bucket/azureblob.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	return len(data), nil
+}
+
+// NewAzureBlobBucket new azure blob bucket
+func NewAzureBlobBucket(accountName, accountKey, containerName, blobServiceBaseUrl, apiVersion string,
+	useHttps bool, directory string) (Bucket, error) {
+	client, err := storage.NewClient(accountName, accountKey,
+		blobServiceBaseUrl, apiVersion, useHttps)
+	if err != nil {
+		log.Fatal(err)
```
Comment (Lcch) [inline bucket/azureblob.go]: done

File: bucket/azureblob.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+		container: b.GetContainerReference(containerName),
+		directory: directory,
+	}, nil
+}
+
+func (bk *AzureBlobBucket) OpenRead(name string) (ObjectReader, error) {
+	blob := bk.container.GetBlobReference(bk.directory + "/" + name)
+	ioReader, err := blob.Get(&storage.GetBlobOptions{})
+	if err != nil {
+		log.Fatal(err)
```
Comment (Lcch) [inline bucket/azureblob.go]: done

File: bucket/bucket.go (hunk 1 above)
Comment (Lcch) [inline bucket/bucket.go]: done

**Pull request merged**